### PR TITLE
feat(markdown-parser): inline parsing and code block parsing

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR implements the inline parsing of emphasis and strong tokens (`*`, `**`, `_`, `__`).

There are still some limitations that I flagged as TODO, but I wanted to roll out the first version because it was getting quite complicated. 

The logic of these tokens is not that easy, here's the documentation: https://github.github.com/gfm/#emphasis-and-strong-emphasis

I tried to put a lot of comments, trying to explain every single choice. The logic is really raw but it does its job.

Also, I put parsing for code blocks but it still needs some polishing that I would like to review later. 

I intentionally turned off formatting and linting, because first time I started enabling the formatter, there were issues with my IDE and the formatter itself around the end line of the file. Because of this, I decided to not move forward for now and review it later once I get a better grasp of the `Builder` utility.

## Test Plan

Updated the smoke test.

### CODE REVIEW

File: scripts/lint-create-rule.ts [hunk 1]
```diff
@@ -76,6 +77,7 @@ export async function main([ruleName]: Array<string>): Promise<number> {
 				key: lint-rules/${ruleName}
 				parent: lint-rules
 				title: ${ruleName}
+				
 			---
```
Comment (sebmck) [inline scripts/lint-create-rule.ts:81]: Why the additional newline here?

**Comment by sebmck** [comment]
Looks good! Only a couple of nits, nothing blocking. Feel free to merge when you're comfortable!

File: scripts/lint-create-rule.ts (hunk 1 above)
Comment (ematipico) [reply scripts/lint-create-rule.ts:81]: Maybe it's not needed now but I was thinking that, format wise, having a space between headings would be something that we want when we will support markdown.

File: internal/markdown-parser/parser/inline.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+				rightFlankingDelimiterFound = canBeRightFlankingDelimiter({
+					startIndex: indexToCheck,
+					endIndex,
+					input: parser.input,
+				});
+
+				// we should stop only if the characters that we found are a right flanking delimiter
+				return !rightFlankingDelimiterFound;
+			},
```
[the remark this answers is not in the document]
Comment (ematipico) [reply internal/markdown-parser/parser/inline.ts:155]: Yes I think we can do it! The function would need some more parameter but that's doable, I will do it

**Pull request merged**